src/EventHandler/ConfigurableEventHandlerEntityBundle.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\EventHandler\ConfigurableEventHandlerEntityBundle.
+ */
+
+namespace Drupal\rules\EventHandler;
+
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Exposes the bundle of an entity as event setting.
+ */
+class ConfigurableEventHandlerEntityBundle extends ConfigurableEventHandlerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function determineQualifiedEvents(Event $event, $event_name, array &$event_definition) {
+    $events_suffixes = [];
+    if ($event instanceof \Drupal\rules\Event\EntityEvent) {
+      $events_suffixes[] = $event->getSubject()->bundle();
+    }
+    return $events_suffixes;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    // Nothing to do by default.
```

Comment:
Do you think we should store event_name in configuration?

in d7 entityInfo is stored in $entityInfo property. looks strange Should I do the same?

You offered to extend Drupal\Component\Plugin\PluginBase in ConfigurableEventHandlerEntityBundle

Coustruct has:
function __construct(array $configuration, $plugin_id, $plugin_definition) {
    parent::__construct($configuration, $plugin_id, $plugin_definition);
  }
what should I put in there? There should be $event and configuration instead?